streams/src/main/java/org/apache/kafka/streams/processor/internals/assignment/Graph.java

```diff
@@ -103,8 +104,21 @@ public String toString() {
         }
     }
 
-    private final SortedMap<V, SortedMap<V, Edge>> adjList = new TreeMap<>();
-    private final SortedSet<V> nodes = new TreeSet<>();
+    private class KeyComparator implements Comparator<V> {
+
+        @Override
+        public int compare(final V o1, final V o2) {
+            if (o1 == null || o2 == null) {
+                return -1;
```

Comment:
If one of both is `null`, why is `o1` considered smaller than `o2`? Would it not be better to define `null` as either smallest or largest globally? -- Also, should we return zero if both are `null`?

Reply:
Good point, if both are `null`, we can return 0;